platform/api.visual/src/org/netbeans/modules/visual/anchor/DefaultAnchorShapeResolver.java

```diff
@@ -49,33 +49,23 @@ public DefaultAnchorShapeResolver(ConnectionWidget connection,
         this.attachedEnd = attachedTo;
         this.attachedWidget = attachedWidget;
     }
-    
-    public int getEndLocation()
-    {
-        int retVal = 0;
-        
-        if(attachedWidget != null)
-        {
+
+    public int getEndLocation() {
+        int retVal;
+
+        if (attachedWidget != null) {
             Rectangle bounds = attachedWidget.getBounds();
             retVal = getLocation(bounds);
-        }
-        else
-        {
-//            Rectangle bounds = attachedWidget.getBounds();
-            Rectangle bounds = new Rectangle();
-            if(attachedEnd == ConnectionEnd.SOURCE)
-            {
-                Widget sourceWidget = getConnection().getSourceAnchor().getRelatedWidget();
-                bounds = sourceWidget.getBounds();
-            }
-            else
-            {
-                Widget sourceWidget = getConnection().getTargetAnchor().getRelatedWidget();
-                bounds = sourceWidget.getBounds();
+        } else {
+            Widget sourceWidget;
+            if (attachedEnd == ConnectionEnd.SOURCE) {
+                sourceWidget = getConnection().getSourceAnchor().getRelatedWidget();
+            } else {
+                sourceWidget = getConnection().getTargetAnchor().getRelatedWidget();
             }
-            retVal = getLocation(bounds);
+            retVal = getLocation(sourceWidget.getBounds());
         }
-        
+
```

Comment:
please don't change the formatting of big blocks. This feels unnecessary and obfuscates the actual change here.

Reply:
reverted formatting below